### PR TITLE
Add default etor filters to UP

### PR DESCRIPTION
**Pull request by jimmyfagan**
This PR adds a framework for having topic-dependent default filters in the UP. I added two maps to `FHIRRouter.kt` which map a topic to a filter, and edited some of the related files to use those maps. I made some adjustments to tests as part of these changes as well.

As a starting point, the etor-ti default quality filter just checks that the message type exists, and the etor-ti default proc mode filter is the same as the existing one for full-elr. That may be as specific as we want to go for now, but we can easily make adjustments in the future, and this provides a pattern that other eventual topics can follow.

Test Steps:
1. Edit the `0149-etor.yml` file to change all instances of `topic: etor-ti` to `topic: full-elr`. Comment out the existing quality filter. Reload those settings by running `./prime multiple-settings set --input ./settings/staging/0149-etor.yml `.
2. Send a lab order by using this postman collection: 
[Temp Collection.postman_collection_json.txt](https://github.com/CDCgov/prime-reportstream/files/11751448/Temp.Collection.postman_collection_json.txt)
3. Query the submission history for the resulting report ID until you see that the report was filtered out by the default quality filter (for the full-elr topic).
4. Edit the `0149-etor.yml` file to change all instances of `topic: full-elr` to `topic: etor-ti`. Leave the existing quality filter commented out. Reload those settings by running `./prime multiple-settings set --input ./settings/staging/0149-etor.yml `.
5. Again send the lab order with the same postman collection as above.
6. Query the submission history for the resulting report ID until you see that the report was delivered to the simulated-lab receiver, meaning it passed the default quality filter for the etor-ti topic.

## Changes
- Changed default quality and proc mode filters to be represented in maps
- Adjusted tests and `isDefaultFilter`/`getQualityFilters`/`getProcessingModeFilter` functions to use the maps

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] Added tests?

## Linked Issues
- Addresses #9464

### CODE REVIEW

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt [hunk 1]
```diff
@@ -83,17 +83,35 @@ class FHIRRouter(
...
     /**
-     * Default Rule:
+     * Default Rules for quality filter on ETOR_TI topic:
+     *   Must have message ID
+     */
+    private val etorTiQualityFilterDefault: ReportStreamFilter = listOf(
+        "%messageId.exists()",
+    )
+
+    val qualityFilterDefaults = mapOf(
```
Comment (luis-pabon-tf) [inline prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt:97]: Could use a lil kdoc just saying why these are mapped

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt [hunk 2]
```diff
@@ -83,17 +83,35 @@ class FHIRRouter(
...
+    /**
+     * Default Rule (used for ETOR_TI and FULL_ELR):
      *  Must have a processing mode id of 'P'
      */
-    val processingModeFilterDefault: ReportStreamFilter = listOf(
+    private val processingModeFilterDefault: ReportStreamFilter = listOf(
         "%processingId.exists() and %processingId = 'P'"
     )
 
+    val processingModeDefaults = mapOf(
```
Comment (luis-pabon-tf) [inline prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt:110]: Like above, just a lil comment for clarity

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt [hunk 3]
```diff
@@ -352,7 +370,7 @@ class FHIRRouter(
                 report,
                 receiver,
                 ReportStreamFilterType.PROCESSING_MODE_FILTER,
-                false
+                defaultResponse = true
```
Comment (luis-pabon-tf) [inline prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt:373]: Just wanted to verify the rationale for this change. Are we sure it doesn't bring unexpected changes?

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt [hunk 4]
```diff
@@ -447,9 +465,12 @@ class FHIRRouter(
...
     internal fun isDefaultFilter(filterType: ReportStreamFilterType, filter: ReportStreamFilter): Boolean {
         // The usage of === (referential equality operator) below is intentional and necessary; we only want to
-        // return true if the filter references the default filter, not if it happens to be equivalent to the default
-        return (filterType == ReportStreamFilterType.QUALITY_FILTER && filter === qualityFilterDefault) ||
-            (filterType == ReportStreamFilterType.PROCESSING_MODE_FILTER && filter === processingModeFilterDefault)
+        // return true if the filter references a default filter, not if it only happens to be equivalent to the default
+        return when (filterType) {
+            ReportStreamFilterType.QUALITY_FILTER -> qualityFilterDefaults.values.any { filter === it }
+            ReportStreamFilterType.PROCESSING_MODE_FILTER -> processingModeDefaults.values.any { filter === it }
+            else -> false
```
Comment (luis-pabon-tf) [inline prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt:472]: love this change, a lot more readable!

**Review by luis-pabon-tf — APPROVED** [review]

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt (hunk 2 above)
Comment (jimmyfagan) [reply prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt:116]: Added!

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**